content/en/docs/concepts/architecture/garbage-collection.md

```diff
@@ -166,17 +166,25 @@ is restarted, the tracked image age is reset, causing the kubelet to wait the fu
 based on image age.
 {{< /note>}}
 
-### Container garbage collection {#container-image-garbage-collection}
+### Container garbage collection {#container-image-garbage-collection} (deprecated)
+
+{{< caution >}}
+Container garbage collection is deprecated in favour of `--eviction-hard` or
+`--eviction-soft` and will be removed in a future version. For more details on
+eviction policy, see [node-pressure-eviction](/docs/concepts/scheduling-eviction/node-pressure-eviction/).
+{{< /caution >}}
 
 The kubelet garbage collects unused containers based on the following variables,
 which you can define:
 
-* `MinAge`: the minimum age at which the kubelet can garbage collect a
-  container. Disable by setting to `0`.
-* `MaxPerPodContainer`: the maximum number of dead containers each Pod 
-  can have. Disable by setting to less than `0`.
+* `MinAge`: the minimum age at which the kubelet can garbage collect a container.
+  Disable by setting a kubelet config flag `--minimum-container-ttl-duration` to `0`.
+* `MaxPerPodContainer`: the maximum number of dead containers each Pod can have.
+  Disable by setting a kubelet config flag `--maximum-dead-containers-per-container`
+  to less than `0`.
 * `MaxContainers`: the maximum number of dead containers the cluster can have.
-  Disable by setting to less than `0`.
+  Disable by setting a kubelet config flag `--maximum-dead-containers`
+  to less than `0`.
```

Comment:
Is this right?
```suggestion
The kubelet garbage collects unused containers based on the values for several
related command line arguments, which are all deprecated:

`--minimum-container-ttl-duration`
: the minimum age at which the kubelet can garbage collect a  container. Disable by setting to 0. Setting a value for `--minimum-container-ttl-duration` is deprecated.

`--maximum-dead-containers`
: limits the total number of dead containers for this node. The default is unlimited. You can set this to an integer that is zero or higher in order to set legacy dead container limits, node-wide. Setting a value for `--maximum-dead-containers` is deprecated.

`--maximum-dead-containers-per-container`: the maximum number of dead containers each Pod 
  can have. The default value for this setting is 1; however, you can specify unlimited dead containers by setting this value to 0. Note that setting a value for `--maximum-dead-containers-per-container` is deprecated.
```

Reply:
yeah I agree we should refer folks to the kubelet configuration options as we're redirecting folks away from using the CLI

Reply:
WDYT ^ @T-Lakshmi ?

Reply:
@haircommander,
Yes, I also agree to mention about CLI usage of these arguments.

But my concern is the below statement is true or not?

 > Container garbage collection is deprecated in favour of `--eviction-hard` or `--eviction-soft` and will be removed in a future version.

also when container GC got deprecated and what are the recommended way to use as alternate to container GC?

Reply:
Also [here](https://kubernetes.io/docs/concepts/scheduling-eviction/node-pressure-eviction/#deprecated-kubelet-garbage-collection-features) it mentioned old logs are stored outside of container's context.

Is that true? if true, in which location old logs are storing now?
Please share your inputs.